EarthAsDMAnalyzer/plugins/EarthAsDMAnalyzer.cc

```diff
@@ -549,22 +541,24 @@ void EarthAsDMAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup&
     
     if (tofMap.isValid()) {
       const reco::MuonTimeExtra* combinedTimeExtra = NULL;
+
       combinedTimeExtra = &tofMap->get(muon.key());
-      muon_tofMap_found_.push_back(1);
-
-      muon_comb_ndof_.push_back( combinedTimeExtra->nDof());
-      muon_comb_timeAtIpInOut_.push_back( combinedTimeExtra->timeAtIpInOut());
-      muon_comb_timeAtIpInOutErr_.push_back( combinedTimeExtra->timeAtIpInOutErr());
-      muon_comb_timeAtIpOutIn_.push_back( combinedTimeExtra->timeAtIpOutIn());
-      muon_comb_timeAtIpOutInErr_.push_back( combinedTimeExtra->timeAtIpOutInErr());
-      muon_comb_invBeta_.push_back( combinedTimeExtra->inverseBeta());
-      muon_comb_freeInvBeta_.push_back( combinedTimeExtra->freeInverseBeta());
-      // Sign convention for muonCombinedFreeInvBeta_:
-      //   positive - outward moving particle
-      //   negative - inward moving particle
-      if (verbose_ > 4) LogPrint(MOD) << "combinedTimeExtra->nDof() " << combinedTimeExtra->nDof() << "  timeAtIpInOut_ " << combinedTimeExtra->timeAtIpInOut()  << " +/- " <<  combinedTimeExtra->timeAtIpOutInErr() << " timeAtIpOutIn_ " << combinedTimeExtra->timeAtIpOutIn() << " +/- " <<  combinedTimeExtra->timeAtIpInOutErr() << " invBeta_ " << combinedTimeExtra->inverseBeta() << " freeInvBeta_ " << combinedTimeExtra->freeInverseBeta();
-    }
-    
+      if (combinedTimeExtra != NULL) {
+        muon_tofMap_found_.push_back(1);
+
+        muon_comb_ndof_.push_back( combinedTimeExtra->nDof());
+        muon_comb_timeAtIpInOut_.push_back( combinedTimeExtra->timeAtIpInOut());
+        muon_comb_timeAtIpInOutErr_.push_back( combinedTimeExtra->timeAtIpInOutErr());
+        muon_comb_timeAtIpOutIn_.push_back( combinedTimeExtra->timeAtIpOutIn());
+        muon_comb_timeAtIpOutInErr_.push_back( combinedTimeExtra->timeAtIpOutInErr());
+        muon_comb_invBeta_.push_back( combinedTimeExtra->inverseBeta());
+        muon_comb_freeInvBeta_.push_back( combinedTimeExtra->freeInverseBeta());
```

Comment:
@LiamBrennan-UCSB I gave this a bit more thought, and I think we should do this in two steps:
 * define the `float nDof = 9999`
 * then if `if (combinedTimeExtra != NULL)` set `nDof = combinedTimeExtra->nDof()`
 * then outside the if statement then fill `nDof`
 * repeat for the rest of them too
--> like this there will be something fill for each event even if the `combinedTimeExtra` is not valid. Can you make a  change and commit that?